src/main/kotlin/com/msd/robot/domain/Robot.kt

```diff
@@ -165,7 +167,10 @@ class Robot(
             UpgradeType.ENERGY_REGEN -> energyRegenLevel++
             UpgradeType.MAX_ENERGY -> energyLevel++
             UpgradeType.HEALTH -> healthLevel++
-            UpgradeType.STORAGE -> storageLevel++
+            UpgradeType.STORAGE -> {
+                storageLevel++
+                inventory.maxStorage= storageByLevel[storageLevel]
```

Comment:
This should happen in the storage level setter, since it needs to happen everytime the storage level gets changed